src/renderer/uia/UiaRenderer.cpp

```diff
@@ -223,6 +234,26 @@ CATCH_RETURN();
 // Return Value:
 // - S_OK, else an appropriate HRESULT for failing to allocate or write.
 [[nodiscard]] HRESULT UiaEngine::EndPaint() noexcept
+{
+    return S_OK;
+}
+
+// RenderEngineBase defines a WaitUntilCanRender() that sleeps for 8ms to throttle rendering.
+// But UiaEngine is never the only engine running. Overriding this function prevents
+// us from sleeping 16ms per frame, when the other engine also sleeps for 8ms.
+void UiaEngine::WaitUntilCanRender() noexcept
+{
+}
+
+// Routine Description:
+// - Used to perform longer running presentation steps outside the lock so the
+//      other threads can continue.
+// - Not currently used by UiaEngine.
+// Arguments:
+// - <none>
+// Return Value:
+// - S_FALSE since we do nothing.
+[[nodiscard]] HRESULT UiaEngine::Present() noexcept

```

Comment:
The diff is weird here. I basically just moved all of the contents of `EndPaint()` over to `Present()`. This way we're not doing this under lock. Leonard said that's the main difference between the two functions.

Reply:
It's also necessary in this case, because `RaiseNotificationEvent` calls back into us (`UiaTextRangeBase::GetAttributeValue` specifically) which locks the console. This only works if we don't hold the lock while calling `RaiseNotificationEvent`.